implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdMeterRegistry.java

```diff
@@ -237,20 +234,30 @@ private void prepareUdpClient(Publisher<String> publisher) {
     }
 
     private void prepareTcpClient(Publisher<String> publisher) {
-        TcpClient.create()
+        AtomicReference<TcpClient> tcpClientReference = new AtomicReference<>();
+        TcpClient tcpClient = TcpClient.create()
                 .host(statsdConfig.host())
                 .port(statsdConfig.port())
                 .handle((in, out) -> out
-                        .options(NettyPipeline.SendOptions::flushOnEach)
                         .sendString(publisher)
                         .neverComplete())
-                .connect()
-                .subscribe(client -> {
-                    this.client.replace(client);
+                .doOnDisconnected(connection -> connectAndSubscribe(tcpClientReference.get()));
+        tcpClientReference.set(tcpClient);
+        connectAndSubscribe(tcpClient);
+    }
 
-                    // now that we're connected, start polling gauges and other pollable meter types
-                    startPolling();
-                });
+    private void connectAndSubscribe(TcpClient tcpClient) {
+        if (started.get()) {
+            tcpClient
+                    .connect()
+                    .retryBackoff(Long.MAX_VALUE, Duration.ofSeconds(1), Duration.ofMinutes(1))
```

Comment:
This ensures that when trying to connect (either initially when the registry is started or later as a reconnect upon being disconnected), it will continue trying to connect to the backend with a backoff delay.

Reply:
not sure if this is a good idea, in the case where `started` is flipped to `false` _between_ two retries...

one way around that would be to wrap the test-and-connect in a `Mono.defer()`:

```java
Mono.defer(() -> {
    if (started.get()) {
        return tcpClient.connect();
    }
    return Mono.empty();
}
    .retryBackoff(...)
```
That way, if the registry is closed after the disconnect, it will correctly stop the retry attempts.

Reply:
Thanks for catching this. I've updated the implementation and added a test for this.